### PR TITLE
gh-107689: Add docstring to the ctypes array object

### PR DESCRIPTION
**Pull request by kostyafarber**
Add a docstring to the ctypes array object. Docs from [here](https://docs.python.org/3.13/library/ctypes.html#ctypes.Array:~:text=Abstract%20base%20class%20for%20arrays.) were used.

* Issue: gh-107689

### CODE REVIEW

File: Misc/NEWS.d/next/Documentation/2023-08-06-15-34-11.gh-issue-107689.oL193x.rst
Comment (sunmy2019) [inline Misc/NEWS.d/next/Documentation/2023-08-06-15-34-11.gh-issue-107689.oL193x.rst]:
```suggestion
Add a docstring to :class:`ctypes.Array`.
```

File: Modules/_ctypes/_ctypes.c
Comment (sunmy2019) [inline Modules/_ctypes/_ctypes.c]: Can you make the line width within 80? This is common practice for code.

Comment (kostyafarber) [reply Modules/_ctypes/_ctypes.c]: No worries will change.

**Comment by kostyafarber** [comment]
@sunmy2019 done

File: Modules/_ctypes/_ctypes.c
Comment (CharlieZhao95) [inline Modules/_ctypes/_ctypes.c]: We can directly use this variable.
```suggestion
    array_doc,                                  /* tp_doc */
```

Comment (kostyafarber) [reply Modules/_ctypes/_ctypes.c]: Ah okay, fair enough. That makes sense.

**Comment by CharlieZhao95** [comment]
> @CharlieZhao95 any updates on this?

Thanks! Just a little reminder. The docstrings I've seen usually describe functions in the form of "what does this function can do", just like:
```
function A
Return True if xxx. If xxx, return False.
```
And it's rare to see "how can we use this function/type":
```
You could subclass type xxx.
```
This is probably not a problem that needs to be modified, I know this docstring comes from the official documentation. Aside from this, LGTM! Next step you could wait for review from core devs :)

**Review by CharlieZhao95 — APPROVED** [review]

**Review by sunmy2019 — APPROVED** [review]
LG! We shall wait for core devs now.

**Review by kumaraditya303 — APPROVED** [review]

**Pull request merged**
